src/index.ts

```diff
@@ -38,19 +38,19 @@ const isObject = (x: unknown): x is object => (
   typeof x === 'object' && x !== null
 );
 
-// check if frozen
-const isFrozen = (obj: object) => (
+// Even if an object is not frozen, any non-configurable _and_ non-writeable properties
+// (which Object.freeze does also do) will break the proxy get trap.
+// See: https://github.com/dai-shi/proxy-compare/pull/8
+const isEffectivelyFrozen = (obj: object) => (
   Object.isFrozen(obj) || (
```

Comment:
I don't think we need to check this any more, right?

Reply:
Yep!

I'd initially taken it out as well, but then later put it back b/c I thought maybe you'd wanted it there on purpose as a cheaper 1st check before doing the `Object.getOwnPropertyDescriptors` + `.some` check which would allocate an array + closure. 

But I was really guessing as to your intent, and agree it's not necessary/probably more clear without it.

Reply:
Yes, Object.isFrozen was very cheap...
Now, the issue https://github.com/pmndrs/valtio/issues/519 is back again.
Reverting this doesn't help, because we changed not to `Object.freeze` in https://github.com/pmndrs/valtio/pull/656.

I ended up with #56. This is very complicated. I wish this library would be simpler.